tcl/util.tcl

```diff
@@ -542,3 +542,42 @@ proc qc::not_null {var} {
     upvar $var value
     return [expr {[info exists value] && $value ne "" }]
 }
+
+proc qc::glob_recursive {args} {
+    #| Return names of files that match patterns, including files in sub-directories
+    # Usage: same as glob (see http://www.tcl.tk/man/tcl8.5/TclCmd/glob.htm),
+    # with additional max_depth switch
+    qc::args $args -join -nocomplain -tails -path ? -directory ? -types ? -max_depth 10 -- args
+
+    # Use a single glob call, but extend each pattern passed in with */ prefixes
+    # (eg. *.html becomes *.html */*.html */*/*.html etc.)
+    if { [info exists join] } {
```

Comment:
Do we need a "-join" option ? 
Can responsibility for this feature be pushed up to the caller ?

`qc::glob_recursive some folder foo`
`qc::glob_recursive some/folder/foo`

Does it make sense for this ?
`qc::glob_recursive path1/folder/ path2/folder`